### DISCUSSION TITLE
Glyph size

### DISCUSSION THREAD

**Discussion by gwww** [wezterm/wezterm#2830] [General]
I'm finding that the glyphs are just a touch too big, they do render fine. Is there a way to tune their size? I have read all of https://github.com/wez/wezterm/issues/342. I assume I'm picking up the default `allow_square_glyphs_to_overflow_width = "WhenFollowedBySpace"`. I'm running latest release (not nightly) build.

Aesthetically, I find the glyphs are just a touch too close to the text. I would be nice for the gap between a glyph and a letter to be the same as the gap between two letters, where a letter could be a space as well.

Is maybe an option something such as `allow_square_glyphs_to_overflow_width = "WhenFollowedByGlyph"`. As I type this I'm feeling the pain of how complicated this is! For example, What happens when you have Glyph, Glyph, space or Glyph, Glyph, non-space character? Shudder.

<img width="258" alt="Screen Shot 2022-12-04 at 13 01 07" src="https://user-images.githubusercontent.com/3060128/205507842-86859ed2-7085-439c-b979-1cd8921e93f4.png">

Maybe this is a feature request and I don't yet know it. I admit maybe this is me over rotating on looks. Anyway, I thought I would open this for discussion, and if nothing happens this is still my favourite terminal!

**Comment by wez** [comment]
Did you find https://wezfurlong.org/wezterm/config/lua/wezterm/font_with_fallback.html#manual-fallback-scaling ?
The glyph size is controlled by the font; wezterm merely asks for a size and the font program produces what it thinks the output should be.
That makes this sort of cross-family size matching a PITA because there's no good way to automatically reason about it.

**Reply by gwww** [reply]
Thank you! I ended up with this:
```
config.font = wt.font_with_fallback({
    { family = "Inconsolata", weight = "Regular" },
    { family = "Symbols Nerd Font Mono", scale = 0.80 },
})
```
The Inconsolata I'm using does not have Nerd Fonts in them, so I've been relying on the fallback that comes with WezTerm.

I tried downloading a Nerd Font version of Inconsolata and found that the glyphs were too small! The advantage of the fallback is that I have a scaling option which is not there when the font is the "complete" package all-in-one.

**Reply by gwww** [reply]
One other advantage of the approach I'm using is that the Inconsolata that I'm using is a variable font. If I understand correctly Nerd Fonts don't come in variable fonts, only static.

I don't know if there is a computational disadvantage (or other disadvantage) but this approach has great flexibility!

Thanks again!

**Comment by badloop** [comment]
Reviving this cause I seem to be running into the same issue.  Not sure if there is a good way on the user side to just force specific characters to be resized, but it looks like wezterm isn't rendering glyphs consistently within itself.  Notice this:

![image](https://user-images.githubusercontent.com/10452840/213565544-b5143ba8-e463-4efd-8bb8-814a23268703.png)

The glyph on the left is being rendered from the definition on the right where it is clearly smaller (and I would expect it to be rendered at that size when it is used).

**Reply by wez** [reply]
Take a look at: https://wezfurlong.org/wezterm/config/lua/config/allow_square_glyphs_to_overflow_width.html

**Reply by badloop** [reply]
That fixed it!  Many thanks!
